lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -328,8 +328,9 @@ def explicit_return?(node)
         end
 
         def return_value_assigned?(node)
-          assignment = assignable_node(node).parent
-          assignment&.lvasgn_type?
+          return false unless (assignment = assignable_node(node).parent)
+
+          assignment.ivasgn_type? || assignment.lvasgn_type?
```

Comment:
Could it be that it’s not limited solely to adding instance variables?

```suggestion
          assignment.assignment?
```

Reply:
done